src/check_datapackage/exclude.py

```diff
@@ -70,19 +71,51 @@ def _get_matches(issue: Issue, exclude: Exclude, descriptor: dict[str, Any]) ->
         return False
 
     if exclude.jsonpath:
-        matches.append(_same_jsonpath(issue, exclude.jsonpath, descriptor))
+        matches.append(_jsonpaths_match(issue, exclude.jsonpath))
 
     if exclude.type:
         matches.append(_same_type(issue, exclude.type))
 
     return all(matches)
 
 
-def _same_jsonpath(issue: Issue, jsonpath: str, descriptor: dict[Any, str]) -> bool:
-    fields: list[DescriptorField] = _get_fields_at_jsonpath(jsonpath, descriptor)
-    jsonpaths: list[str] = _map(fields, lambda field: field.jsonpath)
+def _jsonpaths_match(issue: Issue, jsonpath: str) -> bool:
+    test_object = _get_test_object_from_jsonpath(issue.jsonpath)
+    jsonpaths = _get_direct_jsonpaths(jsonpath, test_object)
     return issue.jsonpath in jsonpaths
 
 
 def _same_type(issue: Issue, type: str) -> bool:
     return type == issue.type
+
+
+def _get_test_object_from_jsonpath(jsonpath: str) -> dict[str, Any]:
+    """Builds an object with a property at the given JSON Path location."""
+    fields = jsonpath.removeprefix("$.").split(".")
+    test_object: dict[str, Any] = {}
+    reduce(_set_object_field, fields, test_object)
+    return test_object
+
+
+def _set_object_field(obj: dict[str, Any], field: str) -> dict[str, Any]:
```

Comment:
To make the purpose clearer here, maybe rename to `_set_field_value_to_placeholder()`?